### DISCUSSION TITLE
RecordQuery where doesn't work

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#1958] [Q&A] [answered]
When I select data using RecordQuery, it gives me zero row.

    err = database.App.Dao().RecordQuery(collection).Where(dbx.NewExp("branchId = {:branchId}", dbx.Params{"branchId": request.BranchId})).OrderBy("created DESC").Limit(1).All(&newsRecords)
	if err != nil {
		return c.JSON(http.StatusBadRequest, err.Error())
	}

Below code is working perfectly.

	err = database.App.Dao().RecordQuery(collection).Where(dbx.NewExp("issuerId = {:issuerId}", dbx.Params{"issuerId": "w4bpvv1vjby7djl"})).OrderBy("created DESC").Limit(5).All(&newsRecords)
	if err != nil {
		return c.JSON(http.StatusBadRequest, err.Error())
	}

I don't know why it doesn't work when I select by branchId. BranchId and IssuerId are both Relation field.

![image](https://user-images.githubusercontent.com/44497354/222487840-872f4721-2db5-4e9f-b793-c3905643bfc7.png)

**Comment by ganigeorgiev** [comment, marked as answer]
There is not enough context to help you debug it.

What is `request.BranchId`?

Is `branchId` "single" or "multiple" `relation` field?

Could you provide a more detailed code sample?

---

Some notes:

1. You can use `dbx.HashExp{"branchId": request.BranchId}` as a shorter alternative to the raw `dbx.NewExp(...)`.

2. If `branchId` is "multiple" relation field the above will not work since internally the value is stored as serialized json array (eg. `["id1", "id2"]`) and for that you have to use either `LIKE` expression or [`json_each`](https://www.sqlite.org/json1.html#jeach). This is also true if initially the field was marked as "multiple" and then switched to "single" because the already inserted records will not be updated (this behavior eventually will change in the future).

**Reply by Oggy4th** [reply]
Hello, there @ganigeorgiev

Thanks for answering.

request.BranchId is coming from API json.

branchId was multiple relation field then I changed it to single. After you told I created new record now it's working. I think when edit collection, it doesn't affect old records.

I was so dumb.
